src/github.com/couchbase/sync_gateway/rest/server_context.go

```diff
@@ -632,3 +650,30 @@ func collectAccessRelatedWarnings(config *DbConfig, context *db.DatabaseContext)
 	return []string{}
 
 }
+
+func takeDbOffline(dbctx *db.DatabaseContext) error {
+
+	base.LogTo("CRUD", "Taking Database : %v, offline", dbctx.Name)
```

Comment:
This looks like it will panic if dbctx comes in nil - should probably handle that scenario.